# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/kiss/posix-fixes.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---


This is mainly needed when used with dietlibc - but also more portable
and more secure.

  Rene Rebe <rene@rocklinux.org>

diff -ur kiss-0.21/src/doread.c src.kiss.-1.25906.3396530713/kiss-0.21/src/doread.c
--- kiss-0.21/doread.c	1998-09-25 14:07:09.000000000 +0200
+++ kiss-0.21/doread.c	2003-08-17 23:54:10.000000000 +0200
@@ -17,11 +17,11 @@
 	for (cp = s.str [1]; *cp; cp++)
 	    if (! isupper (*cp))
 		return (warning ("variable must be in upper case"));
-	gets (buf);
+	fgets (buf, FILENAMELEN, stdin);
 	addtoenv (s.str [1], buf);
     }
     else
-	gets (buf);
+	fgets (buf, FILENAMELEN, stdin);
     
     return (0);
 }
diff -ur kiss-0.21/src/runcmd.c src.kiss.-1.25906.3396530713/kiss-0.21/src/runcmd.c
--- kiss-0.21/runcmd.c	1998-09-25 14:07:10.000000000 +0200
+++ kiss-0.21/runcmd.c	2003-08-17 23:11:35.000000000 +0200
@@ -1,5 +1,7 @@
 #include "kiss.h"
 
+#include <getopt.h>
+
 /* define DEBUG_ONE_CMD to get one shot at gdb'ing the child process */
 /* then make a -g version of the command to see and set a trace in the code */
 
